File: python/888 Fair Candy Swap.py
#time limit
class Solution:
    def fairCandySwap(self, A, B):

        a = sum(A)
        b = sum(B)
        
        B= list(sorted(set(B)))
        n = len(B)
        
        A = list(sorted(set(A)))
        
        m = len(A)
        
        ans = [0, 0]
        each = (a - b) / 2
        for i in range(m):
            try:
                idx = B.index(-each + A[i])
            except ValueError:
                idx = -1
            if idx == -1:
                continue
            else:
                ans = [A[i], B[idx]]
                break
        return ans

File: java/888 Fair Candy Swap.java
class Solution {
    public int[] fairCandySwap(int[] A, int[] B) {
        int m = A.length;
        int n = B.length;
        int a = 0;
        int b = 0;
        
        for (int i : A) a += i;
        for (int j: B) b += j;
        int tmp = (a - b) / 2;

        int[] ans = new int[2];
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (A[i] - B[j] == tmp) {
                    ans[0] = A[i];
                    ans[1] = B[j];
                    break;
                }
            }
        }
        return ans;

    }
}
